hadoop-hdds/common/src/main/java/org/apache/hadoop/ozone/common/utils/BufferUtils.java

```diff
@@ -39,8 +39,8 @@ private BufferUtils() {
    * @param totalLen total length of all ByteBuffers
    * @param bufferCapacity max capacity of each ByteBuffer
    */
-  public static ByteBuffer[] assignByteBuffers(long totalLen,
-      long bufferCapacity) {
+  public static ByteBuffer[] assignByteBuffers(int totalLen,
```

Comment:
The `totalLen` and `buffersAllocated` should be long.  We should allow >= 2G chunks.